### PR TITLE
Feature/ls 101 remerge

### PR DESCRIPTION
**Pull request by milomia**

### CODE REVIEW

File: app/views/legal_services/journey/regional_legal_service.html.erb [hunk 1]
```diff
@@ -54,20 +53,21 @@
...
-                                "regional_legal_service",
-                                region,
-                                false,
+                                "region1[]",
+                                region[1],
+                                @journey.current_step.region1.include?(region[1]),
                                 class: "govuk-checkboxes__input",
                                 id: "region_#{region_no_space}"
                               ) %>
+                    <% c=c+1 %>
```
Comment (kennyevil) [inline app/views/legal_services/journey/regional_legal_service.html.erb]: What is ```c```? As far as I can tell, all that happens with it is that it's declared, incremented and then nothing happens with it.

Also, why are you declaring these regions here and adding your own codes? These are NUTS1 regions and we have a model and static data file for them here 

https://github.com/Crown-Commercial-Service/crown-marketplace/blob/master/app/models/nuts1_region.rb

https://github.com/Crown-Commercial-Service/crown-marketplace/blob/master/data/nuts1_regions.csv

You're making this much harder for yourself than it has to be.

File: app/models/legal_services/journey/lot1_regional_service.rb [hunk 2]
```diff
@@ -1,17 +1,11 @@
 module LegalServices
   class Journey::Lot1RegionalService
     include Steppable
-    attribute :lot1_regional_service
-    validate :validate_lot1_regional_service
+    attribute :lot1, Array
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/models/legal_services/journey/lot1_regional_service.rb]: explicity typing attributes cause issue as discussed

File: app/models/legal_services/journey/regional_legal_service.rb [hunk 3]
```diff
@@ -1,17 +1,10 @@
 module LegalServices
   class Journey::RegionalLegalService
     include Steppable
-    attribute :regional_legal_service
-    validate :validate_regional_legal_service
-
+    attribute :region1, Array
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/models/legal_services/journey/regional_legal_service.rb]: explicity typing attributes cause issue as discussed

File: app/models/legal_services/journey/supplier_results.rb [hunk 4]
```diff
@@ -2,7 +2,7 @@ module LegalServices
   class Journey::SupplierResults
     include Steppable
     def next_step_class
-      Journey::DownloadShortlist
+      Journey::SupplierDetails
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/models/legal_services/journey/supplier_results.rb]: using julias code now from ST

[the remark this answers is not in the document]
Comment (milomia) [reply app/models/legal_services/journey/supplier_results.rb]: this is dummy data

File: app/models/legal_services/region.rb [hunk 5]
```diff
@@ -0,0 +1,33 @@
+module LegalServices
+  class Region
+    include StaticRecord
```
[the remark this answers is not in the document]
Comment (milomia) [reply app/models/legal_services/region.rb]: called it region(s) which are a collection of region data

File: app/views/legal_services/journey/regional_legal_service.html.erb (hunk 1 above)
Comment (milomia) [reply app/views/legal_services/journey/regional_legal_service.html.erb]: removed c - was going to index through array but used .each instead

Comment (milomia) [reply app/views/legal_services/journey/regional_legal_service.html.erb]: also now using nut1_region - unaware that it was already in existence

**Pull request merged**